src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,439 @@
+    private static final String TWO_CHAR_PASSWORD_USER = "a1";
+    private static final String THREE_CHAR_PASSWORD_USER = "a23";
+    private static final String FOUR_CHAR_PASSWORD_USER = "as44";
+    private static final String FIVE_CHAR_PASSWORD_USER = "asdf5";
+    private static final String SIX_CHAR_PASSWORD_USER = "a6dfgh";
+    private static final String SEVEN_CHAR_PASSWORD_USER = "asd7ghj";
+    private static final String EIGHT_CHAR_PASSWORD_USER = "8sdfghjk";
+    private static final String START_WITH_SPACE_PASSWORD_USER = " 1dfghjk";
+    private static final int EIGHTEEN_AGE_USER = 18;
+    private static final int ZERO_AGE_USER = 0;
+    private static final int TEN_AGE_USER = 10;
+    private static final int FIFTEEN_AGE_USER = 15;
+    private static final int SEVENTEEN_AGE_USER = 17;
+    private static final int TWENTY_AGE_USER = 20;
+    private static final int TWENTY_ONE_AGE_USER = 21;
+    private static final int FIFTY_AGE_USER = 50;
+    private static RegistrationService registrationService;
+    private User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        user.setLogin(LOGIN_USER);
+        user.setPassword(PASSWORD_USER);
+        user.setAge(AGE_USER);
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        RegistrationException exception = assertThrows(RegistrationException.class, () -> {
+            registrationService.register(null);
+        });
+        assertEquals(RegistrationExceptionMessage.USER_CAN_NOT_BE_NULL, exception.getMessage(),
+                "Test failed actual exception message: " + exception.getMessage()
+                        + " not equal to " + RegistrationExceptionMessage.USER_CAN_NOT_BE_NULL
+        );
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        user.setLogin(null);
+        RegistrationException exception = assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+        assertEquals(RegistrationExceptionMessage.LOGIN_IS_NOT_NULL, exception.getMessage(),
+                "Test failed actual exception message: " + exception.getMessage()
+                        + " not equal to " + RegistrationExceptionMessage.LOGIN_IS_NOT_NULL
+        );
+    }
+
+    @Test
+    void register_emptyLogin_notOk() {
+        user.setLogin(EMPTY_LOGIN_USER);
+        RegistrationException exception = assertThrows(RegistrationException.class, () -> {
+            registrationService.register(user);
+        });
+        assertEquals(RegistrationExceptionMessage.WRONG_LOGIN_MESSAGE, exception.getMessage(),
+                "Test failed actual exception message: " + exception.getMessage()
+                        + " not equal to " + RegistrationExceptionMessage.WRONG_LOGIN_MESSAGE
+        );
+    }
+
+    @Test
+    void register_oneCharLogin_notOk() {
```

Comment:
just create a lists of valid and invalid logins, and leave only 2 tests for positive and negative cases. Use loop with those lists of valid/invalid logins

Reply:
Ok